### ISSUE TITLE
`explicit_self` rule not working?

### ISSUE THREAD

**Issue by NachoSoto** [realm/SwiftLint#4094]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Given this example in the rule:
```swift
struct ABC {
    func f1() {}
    func f2() {
        f1()
    }
}
```

No warnings are produced

##### Complete output when running SwiftLint, including the stack trace and command used
```bash
$ swiftlint
Linting Swift files in current working directory
Linting 'test.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version: `0.48.0`
* Installation method used: Homebrew
* Paste your configuration file:

```yml
opt_in_rules:
  - explicit_self
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using?  `Xcode 14.0 Build version 14A5294e`
* Do you have a sample that shows the issue?
```bash
echo "struct A { func f1() {} func f2() { f1() } }" | swiftlint lint --no-cache --use-stdin --enable-all-rules
```

**Comment by NachoSoto** [comment]
I see now that this is an "analyzer rule". I hadn't heard about those before.
However this isn't working for me, so surely I'm doing something wrong:
```bash
swiftlint analyze
Error: Could not read compiler invocations
```

**Comment by NachoSoto** [comment]
I see this is documented [here](https://github.com/realm/SwiftLint#analyze). Sorry for the noise!

**Comment by jpsim** [comment]
Yeah analyzer rules are harder to set up and take along time to run but they allow you to run static analysis type rules that are quite powerful and accurate. Let me know if you need more help to set this up.

**Comment by NachoSoto** [comment]
Thanks!

**Comment by NachoSoto** [comment]
I wonder if it should be an error to add an invalid rule in `opt_in_rules`? That would have made me realize right away that I was doing something wrong.

**Comment by SimplyDanny** [comment]
> I wonder if it should be an error to add an invalid rule in `opt_in_rules`? That would have made me realize right away that I was doing something wrong.

This is a good idea, indeed! Wrongly places rules in the configuration should be flagged.

**Comment by jpsim** [comment]
> I wonder if it should be an error to add an invalid rule in `opt_in_rules`? That would have made me realize right away that I was doing something wrong.

Ah yes, good call, sorry I missed this comment.
